services/pkgreview/package-rating.service.js

```diff
@@ -0,0 +1,99 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { starRating, metric } = require('../text-formatters')
+const { colorScale } = require('../color-formatters')
+const { nonNegativeInteger } = require('../validators')
+const { BaseJsonService } = require('..')
+
+const pkgReviewColor = colorScale([2, 3, 4])
+
+const schema = Joi.object({
+  rating: [
+    Joi.number()
+      .min(1)
+      .max(5)
+      .optional(),
+    Joi.allow(null),
```

Comment:
Shouldn't this actually be a range of 0-1? I forgot that the API response was a percentage. Also, you can add the `.allow(null)` onto the schema chain, doesn't need to be a separate option. Something like:

```js
    rating: Joi.number()
      .min(0)
      .max(1)
      .required()
      .allow(null),
```
I believe that the `optional` chain would allow for a response without the `rating` key to get through our response validation which I don't think is what we want. IIUC, the `rating` key will always be in the response with a value of `null` in the case of a package that's never had any reviews, or a value between 0-1 (inclusive on the upper end).